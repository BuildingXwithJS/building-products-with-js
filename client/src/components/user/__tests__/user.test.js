// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import UserWrapper, {User} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: new Date(2016, 1, 1, 1, 1, 1, 1),
};

test('# UserWrapper', () => {
  const store = mockStore({});
  const wrapper = shallow(<UserWrapper user={user} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# User', () => {
  const newLogin = 'newLogin';
  const updateUser = ({login}) => expect(login).toBe(newLogin);

  const component = (
    <User
      user={user}
      updateUser={updateUser}
      edit
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test user update
  const app = mount(component);
  // set new username
  app.find('#loginInput').getDOMNode().value = newLogin;
  // click answer button
  app.find('button').simulate('click');
});
