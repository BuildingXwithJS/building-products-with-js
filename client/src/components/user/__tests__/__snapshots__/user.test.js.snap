// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# User 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    User: 
    <input
      defaultValue="test"
      id="loginInput"
      type="text"
    />
    <div
      className="pull-right"
    >
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="panel-body"
  >
    Registration date: 
    Mon Feb 01 2016 01:01:01 GMT+0100
  </div>
</div>
`;

exports[`# UserWrapper 1`] = `
<User
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  updateUser={[Function]}
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": 2016-02-01T00:01:01.001Z,
    }
  }
/>
`;
